### DISCUSSION TITLE
What is the "PR merge branch"?

### DISCUSSION THREAD

**Discussion by kartikdc** [community/community#26819] [Actions]
<p>I have been working on optimising Actions workflows and only today noticed that actions/checkout when called on a PR without any options checks out a merge branch:</p>
<pre><code class="lang-auto">git checkout --progress --force refs/remotes/pull/3048/merge
</code></pre>
<p>It is also mentioned <a href="https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull_request" rel="noopener nofollow ugc">here</a>:</p>
<blockquote>
<p><code>GITHUB_REF</code></p>
<p>PR merge branch <code>refs/pull/:prNumber/merge</code></p>
</blockquote>
<p>Can someone tell me what exactly this is? Intuitively it sounds like it is just merges target branch into the PR branch and runs the Actions. But is there some more documentation I can find for this? I would like to know what happens when there is merge conflict between the target branch and the PR branch.</p>
<p>Thanks.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="kartikdc" data-post="1" data-topic="204593">
<div class="title">
<div class="quote-controls"></div>
 kartikdc:</div>
<blockquote>
<p>Intuitively it sounds like it is just merges target branch into the PR branch and runs the Actions.</p>
</blockquote>
</aside>
<p>Basically yes, though I’d phrase it the other way around. :slightly_smiling_face:</p>
<aside class="quote no-group" data-username="kartikdc" data-post="1" data-topic="204593">
<div class="title">
<div class="quote-controls"></div>
 kartikdc:</div>
<blockquote>
<p>I would like to know what happens when there is merge conflict between the target branch and the PR branch.</p>
</blockquote>
</aside>
<p>The <a href="https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull_request"><code>pull_request</code> event documentation</a> states that the workflow won’t run in that case.</p>

**Comment by kartikdc** [comment]
<p>Thank you very much. I should have been more careful reading that documentation.</p>
